game-core/src/main/java/games/strategy/net/nio/ServerQuarantineConversation.java

```diff
@@ -77,20 +87,30 @@ public Action message(final Serializable serializable) {
         case CHALLENGE:
           @SuppressWarnings("unchecked")
           final Map<String, String> response = (Map<String, String>) serializable;
+          String error = null;
           if (validator != null) {
-            final String error =
-                Optional.ofNullable(
-                        validator.verifyConnection(
-                            challenge,
-                            response,
-                            remoteName,
-                            remoteMac,
-                            channel.socket().getRemoteSocketAddress()))
-                    .orElseGet(() -> PlayerNameValidation.serverSideValidate(remoteName));
-            send(error);
-            if (error != null) {
+            if (response.containsKey(LobbyLoginResponseKeys.FORGOT_PASSWORD)) {
+              send(
+                  forgotPasswordConversation.handle(channel.socket().getInetAddress(), remoteName));
               step = Step.ACK_ERROR;
```

Comment:
The more I see this code the more I feel like this is unneccessarily complex

Reply:
You're telling me :smile: 
This code is, I think, some of the oldest and basically is not touched because it's brittle, complex, and very important.

Reply:
> brittle, complex, and very important

Legacy Code in a nutshell 😄